tests/tests_integration/test_api/test_workflows.py

```diff
@@ -0,0 +1,348 @@
+from __future__ import annotations
+
+import time
+
+import pytest
+
+from cognite.client import CogniteClient
+from cognite.client.data_classes import Function
+from cognite.client.data_classes.workflows import (
+    CDFTaskParameters,
+    FunctionTaskParameters,
+    TransformationTaskParameters,
+    Workflow,
+    WorkflowCreate,
+    WorkflowDefinitionCreate,
+    WorkflowExecutionList,
+    WorkflowList,
+    WorkflowTask,
+    WorkflowVersion,
+    WorkflowVersionCreate,
+    WorkflowVersionList,
+)
+from cognite.client.exceptions import CogniteAPIError
+
+
+@pytest.fixture
+def workflow_list(cognite_client: CogniteClient) -> WorkflowList:
+    workflow1 = WorkflowCreate(
+        external_id="integration_test-workflow1",
+        description="This is  workflow for testing purposes",
+    )
+    workflow2 = WorkflowCreate(
+        external_id="integration_test-workflow2",
+        description="This is  workflow for testing purposes",
+    )
+    workflows = [workflow1, workflow2]
+    listed = cognite_client.workflows.list()
+    existing = {w.external_id for w in listed}
```

Comment:
Mapping already exist just fyi
```suggestion
    existing = listed._external_id_to_item
```